src/components/Projects/WBS/WBSDetail/Task/Task.jsx

```diff
@@ -63,24 +64,73 @@ function Task(props) {
   const activeController = () => {
     props.setControllerId(props.taskId);
     setControllerRow(!controllerRow);
-  }
+  };
 
   const deleteOneTask = (taskId, mother) => {
     props.deleteWBSTask(taskId, mother);
   };
 
+  function assignColorsToInitials(names) {
+    const colorsHex = [
+      '#FF0000', // red
+      '#0000FF', // blue
+      '#00FF00', // green
+      '#FFFF00', // yellow
+      '#FFA500', // orange
+      '#800080', // purple
+      '#FFC0CB', // pink
+      '#00FFFF', // cyan
+      '#FF00FF', // magenta
+      '#008080', // teal
+      '#00FF00', // lime
+      '#A52A2A', // brown
+      '#000000', // black
+      '#FFD700', // gold
+      '#000080', // navy
+      '#800000', // maroon
+      '#808000', // olive
+    ];
+    let colors = {};
+    let colorIndex = {};
+    
+    for (let name of names) {
+      const initials = name
+        .split(' ')
+        .filter((n, index) => index === 0 || index === name.split(' ').length - 1)
+        .map(n => n[0])
+        .join('')
+        .toUpperCase();
```

Comment:
It is duplicated with Line 217 - 224, I would like to extract a util func.

And it is a bit werid to calculate initals multiple times, I would maintain a hashmap so that you won't need to calculate it again in Line 217 - 224.

Reply:
Did this, please review

Reply:
Amazing job! Thanks~